pulsar/consumer_partition.go

```diff
@@ -1133,6 +1138,15 @@ func (pc *partitionConsumer) internalAckList(request *ackListRequest) {
 }
 
 func (pc *partitionConsumer) MessageReceived(response *pb.CommandMessage, headersAndPayload internal.Buffer) error {
+	defer func() {
+		if r := recover(); r != nil {
+			if err, ok := r.(error); ok && err.Error() == "send on closed channel" {
+				pc.log.WithField("panic", r).Debug("panic recovered in MessageReceived")
```

Comment:
If this were to happen it'd be quite bad in my opinion. Do you really want to raise is as a `DEBUG` only?
I would want to know if you're sending on a closed channel in production code. I suggest a `WARNING` here.

Reply:
I removed this part because it should not reach here. If it reached here, it means `MessageReceived` is called on a closed `partitionConsumer`, which does not make sense because the consumer should unregister itself from the connection.

So I'd like to fail fast rather than recovering from the panic and print a warning message.